include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -403,9 +415,33 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
     }
   }
 
+  if(spatial_discretization == SpatialDiscretization::HDIV)
+  {
+    AssertThrow(
+      use_divergence_penalty == false || use_continuity_penalty == false,
+      dealii::ExcMessage(
+        "Penalty terms can not be used in case of Hdiv function spaces. Think about using L2 instead, or deactivate the penatly terms."));
```

Comment:
There is a typo in "penatly". However, I think one does not need the second part of the explanation. I think one will not switch to L2 just to be able to use the penalty terms.

What is the reason that they cannot be used? Why does the code not work with penalty terms?

A maybe better message could be: "The use of penalty terms does not make sense for SpatialDiscretization::HDIV. You have to deactivate them."

Reply:
I'm pretty sure it was you who wrote this message :D 

The penalty terms are zero in the case of hdiv!